Tools/autotest/quadplane.py

```diff
@@ -78,6 +78,147 @@ def get_disarm_delay(self):
     def set_autodisarm_delay(self, delay):
         self.set_parameter("LAND_DISARMDELAY", delay)
 
+    def test_airmode(self):
+        """Check that plane.air_mode turns on and off as required"""
+        self.progress("########## Testing AirMode operation")
+        self.set_parameter("AHRS_EKF_TYPE", 10)
+        self.change_mode('QSTABILIZE')
+        self.wait_ready_to_arm()
+
+        """SPIN_ARM and SPIN_MIN default to 0.10 and 0.15"""
+        """when armed with zero throttle in AirMode, motor PWM should be at SPIN_MIN"""
+        """If AirMode is off, motor PWM will drop to SPIN_ARM"""
+
+        """Make sure that SERVO5 is Motor1 (default)"""
+        Motor1_value = 33
+        if (self.get_parameter('SERVO5_FUNCTION') != Motor1_value):
+            raise PreconditionFailedException("SERVO5_FUNCTION not %d" % Motor1_value)
+
+        """Make sure that flightmode channel is 5 (default)"""
+        default_fltmode_ch = 5
+        if (self.get_parameter("FLTMODE_CH") != default_fltmode_ch):
+            raise PreconditionFailedException("FLTMODE_CH not %d" % default_fltmode_ch)
+
+        """When disarmed, motor PWM will drop to min_pwm"""
+        #min_pwm = self.get_parameter('SERVO5_MIN')
+        """SERVO5_MIN is not respected; it defaults to 1100, but disarmed PWM is 1000"""
+        min_pwm = self.get_parameter("Q_THR_MIN_PWM")
+
+        """set Q_OPTIONS bit AIRMODE"""
+        airmode_option_bit = (1<<9)
+        self.set_parameter("Q_OPTIONS", airmode_option_bit)
+
+        """set  RC8_OPTION to ARMDISARM"""
+        armdisarm_option = 41
+        arm_ch = 8
+        self.set_parameter("RC%d_OPTION" % arm_ch, armdisarm_option)
+
+        """arm with GCS, record Motor1 SPIN_ARM PWM output and disarm"""
+        spool_delay = self.get_parameter("Q_M_SPOOL_TIME") + 0.25
+        self.zero_throttle()
+        self.arm_vehicle()
+        self.progress("Waiting for Motor1 to spool up to SPIN_ARM")
+        self.delay_sim_time(spool_delay)
```

Comment:
Don't ned this if you're just going to wait on the servo channel value; you can increase the timeout if you need to.

Similarly elsewhere.

Reply:
have to wait long enough for the motor to spool all the way up here, or else spin_arm_pwm is incorrect